flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/TaskActions.java

```diff
@@ -30,15 +30,15 @@
 	/**
 	 * Check the execution state of the execution producing a result partition.
 	 *
-	 * @param jobId ID of the job the partition belongs to.
 	 * @param intermediateDataSetId ID of the parent intermediate data set.
 	 * @param resultPartitionId ID of the result partition to check. This
 	 * identifies the producing execution and partition.
+	 * @param producerReadyCallback callback of producer is in one of consumable states.
 	 */
 	void triggerPartitionProducerStateCheck(
-		JobID jobId,
 		IntermediateDataSetID intermediateDataSetId,
-		ResultPartitionID resultPartitionId);
+		ResultPartitionID resultPartitionId,
+		ThrowingRunnable<Exception> producerReadyCallback);
```

Comment:
Instead of adding this callback, we could also let this method return a `CompletableFuture<Void>` which if completed indicates to retrigger the partition request. That way we would not have to pass in the callback which is forwarded to some other place.